Description: Fix memory leak: Missing regfree() after regcomp().
Origin: backport, http://svn.gnucash.org/trac/changeset/22070
Bug-Debian: http://bugs.debian.org/616515
Last-Update: 2012-09-25
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/backend/xml/gnc-backend-xml.c
+++ b/src/backend/xml/gnc-backend-xml.c
@@ -891,6 +891,7 @@
             else if (regexec(&pattern, stamp_start, 0, NULL, 0) == 0)
                 got_date_stamp = TRUE;
 
+            regfree(&pattern);
             g_free(expression);
 
             if (!got_date_stamp) /* Not a gnucash created file after all... */
--- a/src/html/gnc-html.c
+++ b/src/html/gnc-html.c
@@ -179,6 +179,7 @@
             machine = g_strndup( path + match[1].rm_so, match[1].rm_eo - match[1].rm_so );
         }
     }
+    regfree(&compiled_m);
     return machine;
 }
 
